### PR TITLE
8 readme research questions

### PR DESCRIPTION
**Pull request by EmelineDebalme**
svp review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,162 @@
+
+# Title 
+
+## Abstract
+
+Main characters often drive the plot and central themes of a movie. Hence, it is interesting to concentrate our analysis exclusively on the main character and study the attributes of the actors who portray them. The goal of this data analysis project is to gain insight into the casting and viewers’ preferences for main characters within various movie genres.
+
+Firstly, we'll filter the dataset based on key criteria: prominent genre, relevant time period, main character, and pertinent features. Next, we'll conduct two studies on the filtered dataset:
```
Comment (EmelineDebalme) [inline README.md:8]: put this instead of that

File: README.md [hunk 2]
```diff
@@ -1,3 +1,162 @@
...
+
+## Methods
+
+### Data preparation
+
+In the data preparation phase, we conducted a thorough analysis of data types, including the transformation of dates into years, for example. Unused columns were dropped, and the remaining ones were renamed for consistency. 
+
+An inner merge was performed between IMDb and CMU movies, based on film names and years, augmenting the dataset with ratings and IMDb IDs.
+
+A list of main actors was extracted from the IMDb principals dataset, and then merged with both the merged movies dataframe and Kaggle Awards dataset to add features nomination and awards.
```
Comment (armancenouvel) [inline README.md:38]: Rephrase: In data preparation, we analyzed data types, transformed dates into years, dropped unused columns, and ensured column name consistency. We then merged IMDb and CMU movies, incorporating ratings and IMDb IDs, creating the movies dataframe. Extracting main actors from IMDb principals dataset, we merged them with the combined movies dataframe and Kaggle Awards dataset to include nomination and award features.

File: README.md [hunk 3]
```diff
@@ -1,3 +1,162 @@
...
+- **Nominations**: similarly to the awards won but taking only nominations.
+- **Previous roles**: the number of roles the actor played in prior to this role.
+- **Previous roles in the same genre**: the above column filtered to keep only movies of the same genre
+
+This will lead to the creation of our final dataset composed of two data frames.
+
+<u>**Dataframe 1</u>: Movies**
+
+
+All movies have a unique WikiID but can have the same name or the same release year (not both at the same time), hence there are unique combinations of “name + year”
```
Comment (armancenouvel) [inline README.md:57]: Rephrase : Each movie has a unique WikiID but may share the same name or release year (not simultaneously). Therefore, there are unique combinations of "name + year."

File: README.md [hunk 4]
```diff
@@ -1,3 +1,162 @@
...
+|---------|------------|--------------|--------|--------|
+| W_ID1   | name1      | year1        | Genre1 | Rating1|
+| W_ID2   | name1      | year2        | Genre2 | Rating2|
+| W_ID3   | name2      | year1        | Genre3 | Rating3|
+</div>
+
+
+**<u>Dataframe 2</u>: Main actors**
+
+The Wiki ID / IMDB ID can appear in several rows if there is more than one main character but the combination “Wiki ID + Actor ID” is unique.
```
Comment (armancenouvel) [inline README.md:71]: Rephrase : The Wiki ID / IMDb ID may appear in multiple rows if there are multiple main characters, but the combination "Wiki ID + Actor ID" is unique.

File: README.md [hunk 5]
```diff
@@ -1,3 +1,162 @@
...
+This is an example of results and observations we could obtain: 
+
+
+
+<div style="display: flex; justify-content: space-between;">
+    <div style="width: 53%;">
+        <!-- Your text goes here -->
+
+- From year 1 to year 5, we observe diverging trends for the most selected and highest-rated features. We can conclude that the film industry doesn’t seem to choose their main actors depending on what viewers prefer.
+- From year 6 to year 10, we observe converging trends for both curves meaning the film industry tends to take into account the viewer’s preferences when choosing their main actors.
```
Comment (armancenouvel) [inline README.md:109]: Rephrase : Years 1 to 5 show diverging trends in the most selected and highest-rated features, suggesting the film industry doesn't align actor selection with viewer preferences. In contrast, years 6 to 10 reveal converging trends, indicating the industry considers viewer preferences when choosing main actors.

File: README.md [hunk 6]
```diff
@@ -1,3 +1,162 @@
...
+
+- **Emeline**: Explore analysis, Cross-analysis, Machine Learning, Data-story
+
+- **Théo Houle**: Explore analysis, Cross-analysis, Data-story, README
+
+- **Kelan Solomon**: Feature extraction, Data Visualisation, Machine Learning, Feasibility analysis
+
+- **Dimitri Jacquemont**: Explore analysis, Cross-analysis, Machine Learning, Data-story
+
 # Import data
```
Comment (EmelineDebalme) [inline README.md:160]: remove

File: README.md [hunk 7]
```diff
@@ -1,3 +1,162 @@
...
+        <!-- Your text goes here -->
+
+- From year 1 to year 5, we observe diverging trends for the most selected and highest-rated features. We can conclude that the film industry doesn’t seem to choose their main actors depending on what viewers prefer.
+- From year 6 to year 10, we observe converging trends for both curves meaning the film industry tends to take into account the viewer’s preferences when choosing their main actors.
+    </div>
+    <div style="width: 45%;">
+        <img src="images/crossAnalysis.jpg" alt="Example of cross Analysis result we could obtain" style="width: 100%;">
+        <p style="text-align: center;">Figure 1: Cross Analysis</p>
+    </div>
+</div>
```
Comment (DJacquemont) [inline README.md:115]: html pour l'image

File: README.md [hunk 8]
```diff
@@ -1,3 +1,162 @@
...
+## Research Questions
+
+- Are there any predominant characteristics for actors chosen to portray main characters in a specific movie genre and does it evolve over time?
+
+- How does the notoriety of an actor influence his likeliness to be selected for a given role? Do the viewer's preferences influence the casting? How do honors such as Oscars influence the casting? Does the number of roles prior roles? 
+
+
+## Additional datasets
+
+- **IMDb Non-Commercial Datasets**: The datasets include various aspects of movie and TV show data like titles, crew, ratings, and episode details. In this project are used datasets title.basics (basic title information), title.principals (main participants), title.ratings (user ratings), and name.basics (personnel details).
```
Comment (kelansolo) [inline README.md:24]: TV show -> TV shows

Comment (kelansolo) [reply README.md:24]: In this project, we use the .... datasets

File: README.md [hunk 9]
```diff
@@ -1,3 +1,162 @@
...
+Source: https://datasets.imdbws.com/
+
+- **Kaggle Awards Dataset**: This dataset is a scraping of the official Academy Awards listing winners and nominees between 1927 and 2023. A typical row indicates that a given actor was nominated in a given year for a given movie and whether an oscar was won or not.
```
Comment (kelansolo) [inline README.md:27]: ... Academy Awards, listing ...

File: README.md [hunk 10]
```diff
@@ -1,3 +1,162 @@
...
+Source: https://www.kaggle.com/datasets/unanimad/the-oscar-award
+
+## Methods
+
+### Data preparation
+
+In the data preparation phase, we conducted a thorough analysis of data types, including the transformation of dates into years, for example. Unused columns were dropped, and the remaining ones were renamed for consistency.
```
Comment (kelansolo) [inline README.md:34]: ... transformation of dd/mm/yyyy to years, for example.

File: README.md [hunk 11]
```diff
@@ -1,3 +1,162 @@
...
+
+All movies have a unique WikiID but can have the same name or the same release year (not both at the same time), hence there are unique combinations of “name + year”
+
+<div align="center">
+
+| Wiki ID | Movie name | Release year | Genres | Rating |
+|---------|------------|--------------|--------|--------|
+| W_ID1   | name1      | year1        | Genre1 | Rating1|
+| W_ID2   | name1      | year2        | Genre2 | Rating2|
+| W_ID3   | name2      | year1        | Genre3 | Rating3|
```
Comment (DJacquemont) [inline README.md:65]: Wikipedia movie ID | Movie name | Movie release date | Movie box office revenue | Movie runtime | Movie genres | originalTitle | startYear | IMDb_ID | averageRating | numVotes

File: README.md [hunk 12]
```diff
@@ -1,3 +1,162 @@
...
+- **Kaggle Awards Dataset**: This dataset is a scraping of the official Academy Awards listing winners and nominees between 1927 and 2023. A typical row indicates that a given actor was nominated in a given year for a given movie and whether an oscar was won or not.
+Source: https://www.kaggle.com/datasets/unanimad/the-oscar-award
+
+## Methods
+
+### Data preparation
+
+In the data preparation phase, we conducted a thorough analysis of data types, including the transformation of dates into years, for example. Unused columns were dropped, and the remaining ones were renamed for consistency. 
+
+An inner merge was performed between IMDb and CMU movies, based on film names and years, augmenting the dataset with ratings and IMDb IDs.
```
Comment (kelansolo) [inline README.md:36]: film names and years-> movie name and release date

File: README.md [hunk 13]
```diff
@@ -1,3 +1,162 @@
...
+
+The Wiki ID / IMDB ID can appear in several rows if there is more than one main character but the combination “Wiki ID + Actor ID” is unique.
+
+<div align="center">
+
+| Wiki ID | IMDB ID | Actor ID | Order | Release year | Age | Gender | Roles in movies | Roles in drama movies | Awards | Oscars Nominations |
+|---------|---------|----------|-------|--------------|-----|--------|---------------|---------------------|--------|---------------------|
+| W_ID1   | I_ID1   | A_ID1    | 1     | year1        | Age1| Gender1| Nomination1    | NumDramaMovies1     | Awards1| Ethnicity1          |
+| W_ID1   | I_ID1   | A_ID2    | 2     | year1        | Age2| Gender2| NumMovies2    | NumDramaMovies2     | Awards2| Nomination2          |
+| W_ID2   | I_ID2   | A_ID3    | 3     | year2        | Age3| Gender3| NumMovies3    | NumDramaMovies3     | Awards3| Nomination3          |
```
Comment (DJacquemont) [inline README.md:79]: IMDb_ID | ordering | Wikipedia movie ID | Movie name | Movie release date | primaryName | awards | nominations

File: README.md (hunk 2 above)
Comment (kelansolo) [inline README.md:38]: A list of main actors was extracted from the IMDb principals dataset, and the corresponding nominations and rewards were then added from the Kaggle Awards dataset

File: README.md [hunk 14]
```diff
@@ -1,3 +1,162 @@
...
+
+An inner merge was performed between IMDb and CMU movies, based on film names and years, augmenting the dataset with ratings and IMDb IDs.
+
+A list of main actors was extracted from the IMDb principals dataset, and then merged with both the merged movies dataframe and Kaggle Awards dataset to add features nomination and awards.
+
+---
+
+### Feature extraction
+
+From our various dataset we will retain a few selection of features. For the actors we will keep an ID (sourced from the IMDb dataset), his name, age and gender. For the movies, we keep the wikipedia ID, the IMDb ID, the movie’s title, release year, genre(s) and rating.
```
Comment (kelansolo) [inline README.md:44]: From our various dataset -> datasets

Comment (kelansolo) [reply README.md:44]: his name -> their name

Comment (kelansolo) [reply README.md:44]: pas gender?

Comment (kelansolo) [reply README.md:44]: Verifier les features avec le code!

File: README.md [hunk 15]
```diff
@@ -1,3 +1,162 @@
...
+
+On top of that, we have a few additional actor features that we will engineer based on some simple statistics:
```
Comment (kelansolo) [inline README.md:46]: On top of that -> Additionally
we have a few additional-> supplementary

File: README.md (hunk 3 above)
Comment (kelansolo) [inline README.md:57]: WikiId uniquely identifies each entry

Comment (kelansolo) [reply README.md:57]: (enlever le reste)

File: README.md (hunk 4 above)
Comment (kelansolo) [inline README.md:76]: -> The Wiki ID / IMDB ID and the Actor ID uniquely identify each entry

File: README.md [hunk 16]
```diff
@@ -1,40 +1,169 @@
...
+
+All movies have a unique WikiID but can have the same name or the same release year (not both at the same time), hence there are unique combinations of “name + year”
+
+<div align="center">
+
+| Wiki ID | Movie name | Release year | Genres | Rating |
+|---------|------------|--------------|--------|--------|
+| W_ID1   | name1      | year1        | Genre1 | Rating1|
+| W_ID2   | name1      | year2        | Genre2 | Rating2|
+| W_ID3   | name2      | year1        | Genre3 | Rating3|
```
Comment (EmelineDebalme) [inline README.md:65]: add caption "Illustration of what final table would resemble"

File: README.md [hunk 17]
```diff
@@ -1,3 +1,162 @@
...
+| W_ID2   | I_ID2   | A_ID3    | 3     | year2        | Age3| Gender3| NumMovies3    | NumDramaMovies3     | Awards3| Nomination3          |
+
+
+</div>
+
+---
+
+### Feasibility analysis
+
+After carefully checking our merged dataset and dealing with any missing information, it was found that analyzing it is possible. The review showed that Drama is the most common movie genre, which allows us to look into 17,500 movies and a similar number of main actors. By using statistical methods, the aim is to discover patterns and preferences in the data about actors.
```
Comment (kelansolo) [inline README.md:94]: After checking the final datasets, we can conclude that our analysis is feasible. The features can be found within our datasets and there is enough data. We have implemented the option to change the number of main actors that will be analysed, thus growing our dataset if needed. ( A replacer)

File: README.md [hunk 18]
```diff
@@ -1,3 +1,162 @@
...
+To visualize variations and parallels between the two methods, we will illustrate our observations using the following type of plot.
+This is an example of results and observations we could obtain: 
+
+
+
+<div style="display: flex; justify-content: space-between;">
+    <div style="width: 53%;">
+        <!-- Your text goes here -->
+
+- From year 1 to year 5, we observe diverging trends for the most selected and highest-rated features. We can conclude that the film industry doesn’t seem to choose their main actors depending on what viewers prefer.
```
Comment (kelansolo) [inline README.md:114]: we observe -> we would observe 
highest-rated - top-rated
We can conclude -> We could conclude

Comment (kelansolo) [reply README.md:114]: Idem en dessous pour could

File: README.md [hunk 19]
```diff
@@ -1,3 +1,162 @@
...
+
+2. **Statistical analyses**
+
+We could perform a t-test to verify the observations made before. By looking at the p-value we could state if there are significant differences between the two distributions across all years and for specific years. In the previous example, interesting years to look at could be year 4 (highest difference) and year 10 (smallest difference).
+
+## Further analyses
+
+###  Machine Learning approach:  prediction of viewers’ ratings 
+
+Another interesting analysis could be trying to find a machine learning algorithm to predict the viewer’s ratings of a film depending on the characteristics of the main character. To do so, we will try to perform a linear regression on our data. However, it is difficult to predict now if it will be judicious to use this method. If not, we could use other approaches such as neural networks or decision trees. Next, we'll proceed to train the chosen machine learning algorithm using a predefined training set and test it with the testing dataset.
```
Comment (kelansolo) [inline README.md:133]: Delete: However, it is difficult to predict now if it will be judicious to use this method.

Comment (kelansolo) [reply README.md:133]: Delete: If not, we could use other approaches such as neural networks or decision trees

File: README.md [hunk 20]
```diff
@@ -1,3 +1,162 @@
...
+| P3.3   | Cross-Analysis                  | 08/12        |
+| P3.4   | Machine Learning                | 15/12        |
+| P3.5   | Data Story                      | 20/12        |
+| P3.6   | Most Profitable Actor           | if enough time |
+
+</div>
+
+## Organization within the team
+
+- **Armance**: Feature extraction, Data Visualisation, Feasibility analysis, README
```
Comment (kelansolo) [inline README.md:165]: Armance quoi et Emeline quoi?

**Review by kelansolo — COMMENTED** [review]
Kelan review

**Pull request merged**